app/main.py

```diff
@@ -1,13 +1,41 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+
+    return {
+        "line": error["line_number"], "column": error["column_number"],
+        "message": error["text"], "name": error["code"], "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+
+    return {
+        "errors":
+        [
+            {"line": error["line_number"], "column": error["column_number"],
+             "message": error["text"], "name": error["code"],
+             "source": "flake8"}
+            for error in errors
+        ],
+        "path": file_path,
+        "status": "failed"
+        if len(errors) > 0
+        else {
+            "path": file_path,
+            "status": "passed"
+        }
```

Comment:
There is a syntax error in the return statement. The 'else' part of the ternary conditional expression should not return a dictionary directly. Instead, the entire 'if-else' logic should be used to determine the 'status' value only, not the entire dictionary. Consider restructuring the return statement to avoid this error.